scripts/solve_network.py

```diff
@@ -346,6 +346,116 @@ def add_battery_constraints(n):
     define_constraints(n, lhs, "=", 0, "Link", "charger_ratio")
 
 
+def add_RES_constraints(n, res_share):
+    lgrouper = n.loads.bus.map(n.buses.country)
+    ggrouper = n.generators.bus.map(n.buses.country)
+    sgrouper = n.storage_units.bus.map(n.buses.country)
+    cgrouper = n.links.bus0.map(n.buses.country)
+
+    load = (
+        n.snapshot_weightings.generators
+        @ n.loads_t.p_set.groupby(lgrouper, axis=1).sum()
+    )
+
+    rhs = res_share * load
+
+    res_techs = [
+        "solar",
+        "onwind",
+        "offwind-dc",
+        "offwind-ac",
+        "battery",
+        "hydro",
+        "ror",
+    ]
+    charger = ["H2 electrolysis", "battery charger"]
+    discharger = ["H2 fuel cell", "battery discharger"]
+
+    gens_i = n.generators.query("carrier in @res_techs").index
+    stores_i = n.storage_units.query("carrier in @res_techs").index
+    charger_i = n.links.query("carrier in @charger").index
+    discharger_i = n.links.query("carrier in @discharger").index
+
+    # Generators
+    lhs_gen = (
+        linexpr(
+            (n.snapshot_weightings.generators, get_var(n, "Generator", "p")[gens_i].T)
+        )
+        .T.groupby(ggrouper, axis=1)
+        .apply(join_exprs)
+    )
+
+    # StorageUnits
+    lhs_dispatch = (
+        (
+            linexpr(
+                (
+                    n.snapshot_weightings.stores,
+                    get_var(n, "StorageUnit", "p_dispatch")[stores_i].T,
+                )
+            )
+            .T.groupby(sgrouper, axis=1)
+            .apply(join_exprs)
+        )
+        .reindex(lhs_gen.index)
+        .fillna("")
+    )
+
+    lhs_store = (
+        (
+            linexpr(
+                (
+                    -n.snapshot_weightings.stores,
+                    get_var(n, "StorageUnit", "p_store")[stores_i].T,
+                )
+            )
+            .T.groupby(sgrouper, axis=1)
+            .apply(join_exprs)
+        )
+        .reindex(lhs_gen.index)
+        .fillna("")
+    )
+
+    # Stores (or their resp. Link components)
+    # Note that the variables "p0" and "p1" currently do not exist.
+    # Thus, p0 and p1 must be derived from "p" (which exists), taking into account the link efficiency.
+    lhs_charge = (
+        (
+            linexpr(
+                (
+                    -n.links.loc[charger_i].efficiency,
```

Comment:
So, if we want not to include the losses in the share of res, which I guess is the goal of including this exchange, I believe that the following shall apply:
1. the charger shall have coefficient -1: the p of the link represents the power at the bus0. in our case the bus0 represents the bus (bus A) where the battery is connected, hence coefficient should be -1. The bus 1 of the link is instead an inner bus where the store is connected to (bus B).
2. the discharger shall have coefficient "eta" (not 1/eta): in the case of the discharger, the bus0 represents the inner bus of the battery (bus B), whereas the bus1 is bus A. since we need the power that is fed at B, that power is p*eta. Note that the quantity p(charger) - p(discharger) computed should be negative when summed over all timesteps. That should also be easy to test

Reply:
Thanks for the input Davide. I agree. Changed and tested now, and seems to work as intended. <3